code/modules/mechs/equipment/combat_projectile.dm

```diff
@@ -0,0 +1,167 @@
+/obj/item/mech_equipment/mounted_system/projectile/attackby(var/obj/item/O as obj, mob/user as mob)
+	var/obj/item/gun/projectile/automatic/A = holding
+	if(istype(O, /obj/item/crowbar))
+		A.unload_ammo(user)
+		to_chat(user, SPAN_NOTICE("You remove the ammo magazine from the [src]."))
+	if(istype(O, A.magazine_type))
+		A.load_ammo(O, user)
+		to_chat(user, SPAN_NOTICE("You load the ammo magazine into the [src]."))
+
+/obj/item/mech_equipment/mounted_system/projectile/attack_self(var/mob/user)
+	. = ..()
+	if(. && holding)
+		var/obj/item/gun/M = holding
+		return M.switch_firemodes(user)
+
+/obj/item/gun/projectile/automatic/get_hardpoint_status_value()
+	if(!isnull(ammo_magazine))
+		return ammo_magazine.stored_ammo.len
+	else
+		return null
+
+/obj/item/gun/projectile/automatic/get_hardpoint_maptext()
+	if(!isnull(ammo_magazine))
+		return "[ammo_magazine.stored_ammo.len]/[ammo_magazine.max_ammo]"
+	else
+		return 0
+
+//Weapons below this.
+/obj/item/mech_equipment/mounted_system/projectile
+	name = "mounted submachine gun"
+	icon_state = "mech_ballistic"
+	holding_type = /obj/item/gun/projectile/automatic/smg/mech
+	restricted_hardpoints = list(HARDPOINT_LEFT_HAND, HARDPOINT_RIGHT_HAND)
+	restricted_software = list(MECH_SOFTWARE_WEAPONS)
+
+/obj/item/gun/projectile/automatic/smg/mech
+	magazine_type = /obj/item/ammo_magazine/mech/smg_top
+	allowed_magazines = /obj/item/ammo_magazine/mech/smg_top
+	one_hand_penalty = 0
+	has_safety = FALSE
+	manual_unload = FALSE
+	firemodes = list(
+		list(mode_name="semi auto",      burst=1, fire_delay=null, one_hand_penalty=0, burst_accuracy=null, dispersion=null),
+		list(mode_name="3-round bursts", burst=3, fire_delay=null, one_hand_penalty=0, burst_accuracy=list(0,-1,-1),       dispersion=list(0.0, 1.6, 2.4, 2.4), autofire_enabled=0),
+		list(mode_name="short bursts",   burst=5, fire_delay=null, one_hand_penalty=0, burst_accuracy=list(0,-1,-1,-1,-2), dispersion=list(1.6, 1.6, 2.0, 2.0, 2.4), autofire_enabled=0),
+		list(mode_name="full auto",      burst=1, fire_delay=null, burst_delay=1,      one_hand_penalty=0,                 burst_accuracy=list(0,-1,-1,-1,-2), dispersion=list(1.6, 1.6, 2.0, 2.0, 2.4), autofire_enabled=1)
+	)
+
+/obj/item/mech_equipment/mounted_system/projectile/assault_rifle
+	name = "mounted assault rifle"
+	icon_state = "mech_ballistic2"
+	holding_type = /obj/item/gun/projectile/automatic/assault_rifle/mech
+	restricted_hardpoints = list(HARDPOINT_LEFT_HAND, HARDPOINT_RIGHT_HAND)
+	restricted_software = list(MECH_SOFTWARE_WEAPONS)
+
+/obj/item/gun/projectile/automatic/assault_rifle/mech
```

Comment:
Is there anything stopping people picking these up and using them in their bare hands?

Reply:
Technically speaking no, but they're not obtainable in round outside of inside of the mech system items. The mech system items themselves cannot be fired hand held.